presto-main/src/main/java/com/facebook/presto/server/QueryStateInfoResource.java

```diff
@@ -86,42 +86,51 @@ public QueryStateInfoResource(
 
     @GET
     @Produces(MediaType.APPLICATION_JSON)
-    public void getQueryStateInfos(@QueryParam("user") String user,
-            @QueryParam("includeLocalQueryOnly") @DefaultValue("false") boolean isIncludeLocalQueryOnly,
+    public void getQueryStateInfos(
+            @QueryParam("user") String user,
+            @QueryParam("includeLocalQueryOnly") @DefaultValue("false") boolean includeLocalQueryOnly,
+            @QueryParam("includeAllQueries") @DefaultValue("false") boolean includeAllQueries,
+            @QueryParam("includeAllQueryProgressStats") @DefaultValue("false") boolean includeAllQueryProgressStats,
+            @QueryParam("excludeResourceGroupPathInfo") @DefaultValue("false") boolean excludeResourceGroupPathInfo,
+            @QueryParam("queryTextSizeLimit") Integer queryTextSizeLimit,
             @HeaderParam(X_FORWARDED_PROTO) String xForwardedProto,
             @Context UriInfo uriInfo,
             @Context HttpServletRequest servletRequest,
             @Suspended AsyncResponse asyncResponse)
     {
-        if (resourceManagerEnabled && !isIncludeLocalQueryOnly) {
+        if (resourceManagerEnabled && !includeLocalQueryOnly) {
             proxyQueryStateInfo(servletRequest, asyncResponse, xForwardedProto, uriInfo);
         }
         else {
             List<BasicQueryInfo> queryInfos = dispatchManager.getQueries();
+            Optional<Pattern> userPattern = isNullOrEmpty(user) ? Optional.empty() : Optional.of(Pattern.compile(user));
```

Comment:
`Optional.of(Pattern.compile(user));` is now being eagerly evaluated here for every request.  I would try to think about a way to keep it so it's lazily evaluated.

Reply:
How will eager evaluation matter here ? Eventually if the `user` is not empty, then we will use it always in line 110. The previous implementation was actually possibly compiling the regex for every query, now the compilation happens just once

Reply:
Ok - as I think more - in the earlier implementation, if there were no queries running - we would not have evaluated this pattern. I feel that this is an acceptable cost to pay, since depending on the regex it can be high, but most of the times people may just send simple user regexes - which should be ok to compile. So given the variable cost and the narrow cases in which we might compile (if present) and not use this is low, I feel that the eager compile is worth it.

Reply:
My mistake--this is not eager.  I misread this line.